src/realm/util/basic_system_errors.cpp

```diff
@@ -63,7 +63,7 @@ std::string system_category::message(int value) const
         }
     }
 
-#elif (!REALM_ANDROID && _GNU_SOURCE) || (defined(__USE_GNU) && __ANDROID_API__ >= 23)
+#elif (!REALM_ANDROID && _GNU_SOURCE && !defined(__EMSCRIPTEN__)) || (defined(__USE_GNU) && __ANDROID_API__ >= 23)
```

Comment:
Should the `&& !defined(__EMSCRIPTEN__)` be at the top level rather than in only one half of the `||`? I can't figure out what this condition is trying to do since the left and right side of the `||` are checking for similar but different macros.